src/utils/transactions.js

```diff
@@ -166,25 +166,34 @@ export const mapToEthereumTransactions = async (
 };
 
 // TODO: gasToken support
-export const mapTransactionsToTransactionPayload = (transactions: EthereumTransaction[]): TransactionPayload => {
-  let transactionPayload = mapTransactionToTransactionPayload(transactions[0]);
+export const mapTransactionsToTransactionPayload = (
+  chain: Chain,
+  transactions: EthereumTransaction[],
+): TransactionPayload => {
+  let transactionPayload = mapTransactionToTransactionPayload(chain, transactions[0]);
 
   if (transactions.length > 1) {
     transactionPayload = {
       ...transactionPayload,
-      sequentialTransactions: transactions.slice(1).map(mapTransactionToTransactionPayload),
+      sequentialTransactions: transactions.slice(1).map((
+        transaction,
+      ) => mapTransactionToTransactionPayload(chain, transaction)),
     };
   }
 
-  return transactionPayload;
+  return { ...transactionPayload, chain };
 };
 
 // TODO: gas token support
-const mapTransactionToTransactionPayload = (transaction: EthereumTransaction): TransactionPayload => {
+const mapTransactionToTransactionPayload = (
+  chain: Chain,
+  transaction: EthereumTransaction,
+): TransactionPayload => {
+  const { symbol, decimals } = nativeAssetPerChain[chain];
   const { to, value, data } = transaction;
-  const amount = fromEthersBigNumber(value, 18).toFixed();
+  const amount = fromEthersBigNumber(value, decimals).toFixed();
 
-  return { to, amount, symbol: ETH, data, decimals: 18 };
+  return { to, amount, symbol, data, decimals };
```

Comment:
Do we need to set `contractAddress` (to `nativeAssetPerChain` `address`) here as well?